### ISSUE TITLE
[Feature] Automatically provide feedback to improving configuration based on GPU utilization during fine-tuning

### ISSUE THREAD

**Issue by Glavin001** [axolotl-ai-cloud/axolotl#232]
## Problem

When I first started using axolotl I was frequently adjusting the configuration depending on the GPU utilization and model needs. I was recommended on Discord to execute `nvidia-smi` and continue tweaking the config.

With [ChatGPT's help I found a new command to live-update GPU usage](https://sharegpt.com/c/y1jHRSw):

```bash
nvidia-smi --loop=1 --format=csv,noheader,nounits --query-gpu=utilization.gpu,utilization.memory
```

## Proposed Solution

- [ ] Automatically logging GPU utilization and memory usage during training
- [ ] Provide recommendations for config changes (e.g. increase `batch_size`, etc)

**Comment by winglian** [comment]
Much of this can be exposed via torch as well without making external calls to nvidia-smi.

I do like the general idea though to expose this type of data during training and to make recommendations.

**Comment by ittailup** [comment]
This could be taken a step further and automatically adjust based on current set-up. I really like llm-foundry's option to have the setting `device_train_microbatch_size: auto` and auto gradient accumulation. [1](https://www.mosaicml.com/blog/billion-parameter-gpt-training-made-easy) [2](https://github.com/mosaicml/llm-foundry/blob/7b1e4eadcf52f42da20e2faa194eecfc2c8cdf40/scripts/train/README.md?plain=1#L370)

**Comment by winglian** [comment]
the downside might be that changing the batch size on the fly might be incompatible with how we handle packing currently. Just a not to remind myself mostly.

**Comment by mhenrichsen** [comment]
Currently being reported in the logs.
